imports/ui/layouts/App.jsx

```diff
@@ -1,166 +1,94 @@
-import React, { Component } from 'react';
-import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter, Route, Redirect } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import { Meteor } from 'meteor/meteor';
 
 import { Lists } from '../../api/lists/lists.js';
-import UserMenu from '../components/UserMenu.jsx';
-import ListList from '../components/ListList.jsx';
-import LanguageToggle from '../components/LanguageToggle.jsx';
-import ConnectionNotification from '../components/ConnectionNotification.jsx';
-import Loading from '../components/Loading.jsx';
-import ListPageContainer from '../containers/ListPageContainer.jsx';
-import AuthPageSignIn from '../pages/AuthPageSignIn.jsx';
-import AuthPageJoin from '../pages/AuthPageJoin.jsx';
-import NotFoundPage from '../pages/NotFoundPage.jsx';
+import { GlobalStateProvider } from '../state/GlobalStateProvider.jsx';
+import AppContent from './AppContent.jsx';
 
 const CONNECTION_ISSUE_TIMEOUT = 5000;
 
-export default class App extends Component {
-  static getDerivedStateFromProps(nextProps) {
-    // Store a default list path that can be redirected to from "/" when
-    // the list is ready.
-    const newState = { defaultList: null, redirectTo: null };
-    if (!nextProps.loading) {
-      const list = Lists.findOne();
-      newState.defaultList = `/lists/${list._id}`;
-    }
-    return newState;
-  }
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      showConnectionIssue: false,
-      defaultList: null,
-      redirectTo: null,
-    };
-    this.toggleMenu = this.toggleMenu.bind(this);
-    this.closeMenu = this.toggleMenu.bind(this, false);
-    this.logout = this.logout.bind(this);
-  }
+const App = ({
+  // current meteor user
+  user,
+  // server connection status
+  connected,
+  // subscription status
+  loading,
+  // all lists visible to the current user
+  lists,
+}) => {
+  const [showConnectionIssue, setShowConnectionIssue] = useState(false);
+  const [defaultList, setDefaultList] = useState(null);
+  const [redirectTo, setRedirectTo] = useState(null);
 
-  componentDidMount() {
+  useEffect(() => {
     setTimeout(() => {
-      /* eslint-disable react/no-did-mount-set-state */
-      this.setState({ showConnectionIssue: true });
+      setShowConnectionIssue(true);
     }, CONNECTION_ISSUE_TIMEOUT);
-  }
+  }, []);
```

Comment:
Careful, we need to clear the timeout if the component gets unmounted in order to avoid running the callback when there's no need.

```js
useEffect(() => {
  const timeout = setTimeout(() => {
    ...
  }, CONNECTION_ISSUE_TIMEOUT);

  return () => {
    clearTimeout(timeout);
  }
}, []);
```

Reply:
You are right.